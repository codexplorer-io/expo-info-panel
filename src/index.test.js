import React from 'react';
import { shallow } from 'enzyme';
import {
    Button,
    Provider as PaperProvider
} from 'react-native-paper';
import { InfoPanel } from './index';

const renderComponent = props => (
    <PaperProvider>
        <InfoPanel {...props} />
    </PaperProvider>
);

const defaultProps = {
    icon: 'warning',
    text: 'dummy text'
};

describe('InfoPanel', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    describe('Without actions', () => {
        it('should render as expected', () => {
            // eslint-disable-next-line lodash/prefer-lodash-method
            const wrapper = shallow(renderComponent(defaultProps))
                .find(InfoPanel)
                .dive();

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('With actions', () => {
        const onAction1 = jest.fn();
        const onAction2 = jest.fn();
        const props = {
            ...defaultProps,
            actions: [
                {
                    text: 'action 1',
                    action: onAction1
                },
                {
                    text: 'action 2',
                    action: onAction2
                }
            ]
        };

        it('should render as expected', () => {
            // eslint-disable-next-line lodash/prefer-lodash-method
            const wrapper = shallow(renderComponent(props))
                .find(InfoPanel)
                .dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('should execute action 1 onPress', () => {
            // eslint-disable-next-line lodash/prefer-lodash-method
            const onPress = shallow(renderComponent(props))
                .find(InfoPanel)
                .dive()
                .find(Button)
                .at(0)
                .prop('onPress');

            onPress();

            expect(onAction1).toHaveBeenCalledTimes(1);
        });

        it('should execute action 2 onPress', () => {
            // eslint-disable-next-line lodash/prefer-lodash-method
            const onPress = shallow(renderComponent(props))
                .find(InfoPanel)
                .dive()
                .find(Button)
                .at(1)
                .prop('onPress');

            onPress();

            expect(onAction2).toHaveBeenCalledTimes(1);
        });
    });
});
