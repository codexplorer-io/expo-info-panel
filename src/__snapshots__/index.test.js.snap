// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel With actions should render as expected 1`] = `
<Styled(withTheme(Surface))>
  <Styled(View)>
    <Styled(Icon)
      name="warning"
      size={24}
      type="info"
    />
    <Styled(Caption)>
      dummy text
    </Styled(Caption)>
  </Styled(View)>
  <Styled(View)>
    <withTheme(Button)
      key="action 1"
      labelStyle={
        {
          "fontSize": 12,
        }
      }
      mode="text"
      onPress={[MockFunction]}
    >
      action 1
    </withTheme(Button)>
    <withTheme(Button)
      key="action 2"
      labelStyle={
        {
          "fontSize": 12,
        }
      }
      mode="text"
      onPress={[MockFunction]}
    >
      action 2
    </withTheme(Button)>
  </Styled(View)>
</Styled(withTheme(Surface))>
`;

exports[`InfoPanel Without actions should render as expected 1`] = `
<Styled(withTheme(Surface))>
  <Styled(View)>
    <Styled(Icon)
      name="warning"
      size={24}
      type="info"
    />
    <Styled(Caption)>
      dummy text
    </Styled(Caption)>
  </Styled(View)>
</Styled(withTheme(Surface))>
`;
